spectrum_utils/plot.py

```diff
@@ -227,6 +232,111 @@ def spectrum(
     return ax
 
 
+def mass_errors(
+    spec: MsmsSpectrum,
+    *,
+    plot_unknown: bool = True,
+    color_ions: bool = True,
+    grid: Union[bool, str] = True,
+    ax: Optional[plt.Axes] = None,
+) -> plt.Axes:
+    """
+    Plot mass error bubble plot for a given spectrum.
+
+    Parameters
+    ----------
+    spec : MsmsSpectrum
+        The spectrum with mass errors to be plotted.
+    plot_unknown : bool, optional
+        Flag indicating whether or not to plot mass errors for unknown peaks.
+    color_ions : bool, optional
+        Flag indicating whether or not to color dots for annotated fragment
+        ions. The default is True.
+    grid : Union[bool, str], optional
+        Draw grid lines or not. Either a boolean to enable/disable both major
+        and minor grid lines or 'major'/'minor' to enable major or minor grid
+        lines respectively.
+    ax : Optional[plt.Axes], optional
+        Axes instance on which to plot the mass errors. If None the current
+        Axes instance is used.
+
+    Returns
+    -------
+    plt.Axes
+        The matplotlib Axes instance on which the mass errors are plotted.
+
+    Notes
+    -----
+    The mass error bubble plot was first introduced in [1]_.
+
+    References
+    ----------
+    .. [1] Barsnes,H., Eidhammer,I. and Martens,L. (2010)
+       FragmentationAnalyzer: An open-source tool to analyze MS/MS
+       fragmentation data. PROTEOMICS, 10, 1087–1090.
+       doi:10.1002/pmic.200900681
+
+    """
+    if ax is None:
+        ax = plt.gca()
+
+    _format_ax(ax, grid)
+    ax.set_ylabel("mass error")
+
+    if len(spec.mz) == 0:
+        return ax
+
+    round_mz = 50
+    max_mz = math.ceil(spec.mz[-1] / round_mz + 1) * round_mz
+    ax.set_xlim(0, max_mz)
+
+    annotations = (
+        spec.annotation
+        if spec.annotation is not None
+        else itertools.repeat(None)
+    )
+
+    dot_colors = []
+    is_known_ion = []
+    for ann in annotations:
+        # Use the first annotation in case there are multiple options.
+        ion_type = ann[0].ion_type[0] if ann is not None else None
+        is_known_ion.append(ion_type != "?" and ion_type != None)
+        color = colors.get(ion_type if color_ions else None)
+        dot_colors.append(color)
+    dot_colors = np.array(dot_colors)
+
+    mask = (
+        np.ones_like(spec.mz, dtype=bool)
+        if plot_unknown
+        else np.array(is_known_ion)
+    )
+
+    get_mz_delta = np.vectorize(
+        lambda a: a.fragment_annotations[0].mz_delta[0]
+        if a.fragment_annotations
+        else 0.0
+    )
+
+    mz_deltas = get_mz_delta(spec.annotation)
+    max_intensity = spec.intensity.max()
+    intensity_scaled = 500 * (spec.intensity / max_intensity)
+
+    max_abs_error = max(abs(mz_deltas))
+    ax.set_ylim(-max_abs_error, max_abs_error)
+
+    ax.scatter(
+        spec.mz[mask],
+        mz_deltas[mask],
+        s=intensity_scaled[mask],
```

Comment:
**todo:** Can you mention in the docstring that the size of the bubbles is relative to the peak intensity?